src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package taxi.service;
+
+import java.util.Objects;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Can't find driver with this login and password."));
+        if (!Objects.equals(password, driver.getPassword())) {
+            throw new AuthenticationException("Can't find driver with this login and password.");
```

Comment:
rewrite your logic, so you won't need to throw this exception twice in your code